### PR TITLE
Remove support for Async callback types when used in Subscribe functions

### PR DESCRIPTION
**Pull request by dekz**
This PR:
* Removes in: Exchange subscribe, Token subscribe and OrderWatcher subscribe.

### CODE REVIEW

File: packages/0x.js/src/contract_wrappers/contract_wrapper.ts [hunk 1]
```diff
@@ -52,8 +52,7 @@ export class ContractWrapper {
             throw new Error(ZeroExError.SubscriptionNotFound);
         }
         if (!_.isUndefined(err)) {
-            const callback = this._filterCallbacks[filterToken];
-            callback(err, undefined);
+            this._filterCallbacks[filterToken](err, undefined);
```
Comment (fabioberger) [inline packages/0x.js/src/contract_wrappers/contract_wrapper.ts]: I actually prefer us to first set it to a variable called `callback`, for readability.

File: packages/0x.js/src/index.ts [hunk 2]
```diff
@@ -6,8 +6,6 @@ export {
     ECSignature,
     ZeroExError,
     EventCallback,
-    EventCallbackAsync,
-    EventCallbackSync,
```
Comment (fabioberger) [inline packages/0x.js/src/index.ts]: Can you also open a PR in the website repo, removing these from the `public0xjsTypes` array in `utils/constants.ts`.

File: packages/0x.js/CHANGELOG.md [hunk 3]
```diff
@@ -1,13 +1,17 @@
 # CHANGELOG
 
+v0.25.0 - _November 14, 2017_
```
Comment (fabioberger) [inline packages/0x.js/CHANGELOG.md]: Let's just leave this as `vx.x.x`

**Comment by fabioberger** [comment]
💯

File: packages/0x.js/src/index.ts (hunk 2 above)
Comment (LogvinovLeon) [reply packages/0x.js/src/index.ts]: No. We shouldn't remove them cause they're still present in the old versions

**Pull request merged**